### PR TITLE
Add fine-tune

### PR DESCRIPTION
**Pull request by markus-hinsche**
**Motivation**: [Run65](https://ml.azure.com/experiments/id/da5aef2b-b171-44bd-8480-749dcfdd5258/runs/q3-depthmapmultiartifactlatefusion-plaincnn-height-95k_1603272438_1d60b423?wsid=/subscriptions/9b82ecea-6780-4b85-8acf-d27d79028f07/resourceGroups/cgm-ml-prod/providers/Microsoft.MachineLearningServices/workspaces/cgm-azureml-prod&tid=006dabd7-456d-465b-a87f-f7d557e319c8) shows that it can help slightly (see val_mae) to fine-tune with unfrozen weights.

image
Figure: This experiment trains with a frozen base net for 1000 epochs, and then fine-tunes (unfreezes base layers to train them together with the network head).

**Changes:**
* Include fine-tuning as the default configuration

### CODE REVIEW

**Comment by AI-Guru** [comment]
That is very, very interesting! Definitely a great strategy to have in our toolbox.

What do you think about the val MAE plot? Before unfreezing the losses pre-1k steps are below 0.8 and very fluctuating. After 1k steps the loss is less fluctuating but above 0.8. 

How do they differ in evaluation with the respecive dataset?

**Comment by AI-Guru** [comment]
I went through the code. A suggestion.

What about unfreezing either by adding a new callback to the training or by customizing model.fit()? An unfreezing callback would make that finetuning a plug-and-play feature.

**Comment by AI-Guru** [comment]
Go ahead and merge.

**Comment by markus-hinsche** [comment]
I agree, a callback might be a good way to do this. We can consider this when we touch this code again
